seisbench/models/base.py

```diff
@@ -722,12 +841,224 @@ async def _annotate_async(
+            (
+                "reassemble",
+                queue_pred_blocks,
+                parallelism,
+                1,
+            ),  # terminates predictions_to_stream
+            ("predictions_to_stream", queue_results, 1, 1),  # Indicates job completion
+        ]
+        process_watchdog = torchmp.Process(
+            target=_watchdog, args=(queue_watchdog, watchdog_tasks)
+        )
+
+        process_streams_to_arrays = torchmp.Process(
+            target=self._process_streams_to_arrays,
+            args=(
+                queue_watchdog,
+                queue_groups,
+                queue_raw_blocks,
+                strict,
+                flexible_horizontal_components,
+            ),
+        )
+        cut_processes = []
+        pre_processes = []
+        post_processes = []
+        reassemble_processes = []
+        for i in range(parallelism):
+            cut_processes += [
+                torchmp.Process(
+                    target=cut_fragments,
+                    args=(
+                        queue_watchdog,
+                        queue_raw_blocks,
+                        queue_raw_fragments,
+                        argdict,
+                    ),
+                )
+            ]
+            pre_processes += [
+                torchmp.Process(
+                    target=self._process_annotate_window_pre,
+                    args=(
+                        queue_watchdog,
+                        queue_raw_fragments,
+                        queue_preprocessed_fragments,
+                        argdict,
+                    ),
+                )
+            ]
+            post_processes += [
+                torchmp.Process(
+                    target=self._process_annotate_window_post,
+                    args=(
+                        queue_watchdog,
+                        queue_raw_pred,
+                        queues_postprocessed_pred,
+                        argdict,
+                    ),
+                )
+            ]
+            reassemble_processes += [
+                torchmp.Process(
+                    target=reassemble_blocks,
+                    args=(
+                        queue_watchdog,
+                        queues_postprocessed_pred[i],
+                        queue_pred_blocks,
+                        argdict,
+                    ),
+                )
+            ]
+        process_predictions_to_streams = torchmp.Process(
+            target=self._process_predictions_to_streams,
+            args=(queue_watchdog, queue_pred_blocks, queue_results),
+        )
+
+        process_streams_to_arrays.start()
+        for proc in (
+            cut_processes + pre_processes + post_processes + reassemble_processes
+        ):
+            proc.start()
+        process_predictions_to_streams.start()
+
+        for group in groups:
+            queue_groups.put(group)
+
+        process_watchdog.start()  # Why does it cause a race condition?
```

Comment:
If this is still an issue, can we elaborate on this comment or tag it with a TODO:

Reply:
Upps, left that in here. No, the issue was resolve by adding the `target_queue.join()` in the watchdog process. I'll remove it.